diff --git a/src/utils/lombok/javac/Javac.java b/src/utils/lombok/javac/Javac.java
index 6f99463b..2b1b0e56 100644
--- a/src/utils/lombok/javac/Javac.java
+++ b/src/utils/lombok/javac/Javac.java
@@ -324,9 +324,9 @@ public class Javac {
 		if (symtabVoidType != null) try {
 			return (Type) symtabVoidType.get(symbolTable);
 		} catch (IllegalAccessException ignore) {}
-		
+
 		if (Javac.getJavaCompilerVersion() < 8) {
-			return new JCNoType(((Integer) tag.value).intValue());
+			throw new UnsupportedOperationException("os-xtoo lombok does not support java versions before 1.8");
 		} else {
 			try {
 				if (CTC_VOID.equals(tag)) {
@@ -341,25 +341,7 @@ public class Javac {
 			}
 		}
 	}
-	
-	private static class JCNoType extends Type implements NoType {
-		public JCNoType(int tag) {
-			super(tag, null);
-		}
-		
-		@Override
-		public TypeKind getKind() {
-			if (tag == ((Integer) CTC_VOID.value).intValue()) return TypeKind.VOID;
-			if (tag == ((Integer) CTC_NONE.value).intValue()) return TypeKind.NONE;
-			throw new AssertionError("Unexpected tag: " + tag);
-		}
-		
-		@Override
-		public <R, P> R accept(TypeVisitor<R, P> v, P p) {
-			return v.visitNoType(this, p);
-		}
-	}
-	
+
 	private static final Field JCCOMPILATIONUNIT_ENDPOSITIONS, JCCOMPILATIONUNIT_DOCCOMMENTS;
 	static {
 		Field f = null;
