diff --git a/pym/repoman/copyrights.py b/pym/repoman/copyrights.py
--- a/pym/repoman/copyrights.py	2018-04-17 20:34:25.910613600 -0400
+++ b/pym/repoman/copyrights.py	2018-04-17 20:37:03.994433254 -0400
@@ -15,10 +15,9 @@
 
 
 _copyright_re1 = \
-	re.compile(br'^(# Copyright \d\d\d\d)-\d\d\d\d( Gentoo Foundation)\b')
+	re.compile(br'^(# Copyright \d\d\d\d)-\d\d\d\d( (Gentoo Foundation|Obsidian-Studios))\b')
 _copyright_re2 = \
-	re.compile(br'^(# Copyright )(\d\d\d\d)( Gentoo Foundation)\b')
-
+	re.compile(br'^(# Copyright )(\d\d\d\d)( (Gentoo Foundation|Obsidian-Studios))\b')
 
 class _copyright_repl(object):
 	__slots__ = ('year',)
diff --git a/pym/repoman/modules/linechecks/gentoo_header/header.py b/pym/repoman/modules/linechecks/gentoo_header/header.py
--- a/pym/repoman/modules/linechecks/gentoo_header/header.py	2018-04-17 20:34:25.920613209 -0400
+++ b/pym/repoman/modules/linechecks/gentoo_header/header.py	2018-04-17 20:39:28.098799434 -0400
@@ -18,6 +18,7 @@
 	repoman_check_name = 'ebuild.badheader'
 
 	gentoo_copyright = r'^# Copyright ((1999|2\d\d\d)-)?%s Gentoo Foundation$'
+	os_copyright = r'^# Copyright (2\d\d\d-)?%s Obsidian-Studios, Inc\.$'
 	gentoo_license = (
 		'# Distributed under the terms'
 		' of the GNU General Public License v2')
@@ -32,12 +33,15 @@
 			self.modification_year = str(time.gmtime(pkg.mtime)[0])
 		self.gentoo_copyright_re = re.compile(
 			self.gentoo_copyright % self.modification_year)
+		self.os_copyright_re = re.compile(
+			self.os_copyright % self.modification_year)
 
 	def check(self, num, line):
 		if num > 2:
 			return
 		elif num == 0:
-			if not self.gentoo_copyright_re.match(line):
+			if not ( self.gentoo_copyright_re.match(line) or
+				 self.os_copyright_re.match(line) ):
 				return self.errors['COPYRIGHT_ERROR']
 		elif num == 1 and line.rstrip('\n') != self.gentoo_license:
 			return self.errors['LICENSE_ERROR']
